src/shogun/preprocessor/RandomFourierGaussPreproc.h

```diff
@@ -89,16 +80,14 @@ class RandomFourierGaussPreproc: public RandomMixin<DensePreprocessor<float64_t>
 
 	virtual SGMatrix<float64_t> apply_to_matrix(SGMatrix<float64_t> matrix) override;
 
-	virtual SGMatrix<float64_t> sample_spectral_density(int32_t dim_input_space) const;
-
 	/** Helper method which generates random coefficients and stores in the
 	 *  internal members.
 	 *
 	 * @param dim_input_space input space dimension
 	 */
 	virtual void init_basis(int32_t dim_input_space);
 
-	float64_t m_log_width = -0.34657359027997264;
+	std::shared_ptr<ShiftInvariantKernel> m_kernel = std::make_shared<GaussianKernel>();
```

Comment:
does this actually work? Because I thought that inline requires static initialisation, which I am pretty sure you cannot do with shared_ptr..

Reply:
It doesn't throw a compiler error I'll tell you that

Reply:
no default init please ... doenst make sense

Reply:
instead, you should assert that a kernel is present in the fit method